src/main/java/build/buildfarm/server/ActionCacheService.java

```diff
@@ -98,6 +103,7 @@ public void onFailure(Throwable t) {
         },
         directExecutor());
     actionResultsMetric.inc();
+    metricsPublisher.publishRequestMetadata(requestMetadata);
```

Comment:
1.  If we don't configure a metricsPublisher in our server config it defaults to using a log publisher.  That seems okay.  I was thinking there might be an option to disable it entirely, but we can also set a FINE log level in the `LogMetricsConfig` which will do the same thing.  I'm just double checking that this can be no-op if desired.

2. `requestMetadata` is only part of the full action request.  To get the full context you could combine `GetActionResultRequest` + `RequestMetadata`.  Then you'll have additional context about getting results inlined (which could have different performances and success rates, the instance name (which probably isn't that important to you).  I also see the metadata also has the actionId so the extra information from `GetActionResultRequest` might not be that important after all.

Reply:
For 1. Correct - you can just set the log level to FINE and never see the log message. For 2. I'd have to check and see if there is any useful information in the GetActionResultRequest. For my purpose, the RequestMetadata contains everything I need. We may need to balance the size of the data vs useful information.